### PR TITLE
Harden keybinding parsing, modes/conflicts, and chord coverage

### PR DESCRIPTION
**Pull request by RtlZeroMemory**
## Summary
This PR hardens the existing keybinding system without adding new syntax/features. It audits and tightens parsing, chord/mode behavior, and conflict resolution, then adds deterministic regression coverage and updates docs.

## Fixes
- Reject duplicate modifiers in key strings (e.g. `ctrl+ctrl+a`).
- Make same-sequence re-registration replace the prior binding in mode registration flows.
- Make `setMode()` throw for unknown/unregistered modes (clear error).
- Preserve handler-driven mode changes during app key routing so they apply to subsequent events.

## Tests Added
New deterministic suites under `packages/core/src/keybindings/__tests__/`:
- `keybinding.parse.test.ts`
- `keybinding.chord.test.ts`
- `keybinding.modes.test.ts`
- `keybinding.conflicts.test.ts`
- `keybinding.mode-switch.test.ts`

The new suites cover parser edge cases, chord matching/timeout semantics, mode inheritance/switching, routing/conflict precedence, and integration behavior around handler-triggered mode switches.

## Documentation
- Updated `docs/guide/input-and-focus.md` with a consolidated **Keybinding System** section documenting syntax, chord behavior, mode rules, conflict resolution, and routing order.

## Validation
- `npm run lint`
- `npm run build`
- `npm run typecheck`
- `node scripts/run-tests.mjs`

All commands pass on this branch.

### CODE REVIEW

File: packages/core/src/app/createApp.ts [hunk 1]
```diff
@@ -715,8 +738,9 @@ export function createApp<S>(
                   update: app.update,
                   focusedId: widgetRenderer.getFocusedId(),
                 });
-                const keyResult = routeKeyEvent(keybindingState, ev, keyCtx);
-                keybindingState = keyResult.nextState;
+                const routeInputState = keybindingState;
+                const keyResult = routeKeyEvent(routeInputState, ev, keyCtx);
+                applyRoutedKeybindingState(routeInputState, keyResult.nextState);
                 if (keyResult.consumed) {
```
[the remark this answers is not in the document]
Comment (RtlZeroMemory) [reply packages/core/src/app/createApp.ts:751]: Fixed in `220d04c`.

`createApp` now checks `keyResult.handlerError` after keybinding routing (for both raw key events and synthetic key events from text input), enqueues `ZRUI_USER_CODE_THROW`, and stops processing the batch.

Added regression test: `unknown mode switch inside keybinding handler surfaces fatal error` in `packages/core/src/keybindings/__tests__/keybinding.mode-switch.test.ts`.
